samples/dnn/siamrpnpp.py

```diff
@@ -0,0 +1,419 @@
+                 self.window * self.track_window_influence
+        best_idx = np.argmax(pscore)
+        bbox = pred_bbox[:, best_idx] / scale_z
+        lr = penalty[best_idx] * score[best_idx] * self.track_lr
+
+        cx = bbox[0] + self.center_pos[0]
+        cy = bbox[1] + self.center_pos[1]
+
+        # smooth bbox
+        width = self.size[0] * (1 - lr) + bbox[2] * lr
+        height = self.size[1] * (1 - lr) + bbox[3] * lr
+
+        # clip boundary
+        cx, cy, width, height = self._bbox_clip(cx, cy, width, height, img.shape[:2])
+
+        # udpate state
+        self.center_pos = np.array([cx, cy])
+        self.size = np.array([width, height])
+        bbox = [cx - width / 2, cy - height / 2, width, height]
+        best_score = score[best_idx]
+        return {'bbox': bbox, 'best_score': best_score}
+
+def get_frames(video_name):
+    """
+    Args:
+        Path to input video frame
+    Return:
+        Frame
+    """
+    if not video_name:
+        cap = cv.VideoCapture(0)
+        while True:
+            ret, frame = cap.read()
+            if ret:
+                yield frame
+            else:
+                break
+    elif video_name.endswith('avi') or \
+        video_name.endswith('mp4'):
+        cap = cv.VideoCapture(video_name)
+        while True:
+            ret, frame = cap.read()
+            if ret:
+                yield frame
+            else:
+                break
+    else:
+        images = glob(os.path.join(video_name, '*.jp*'))
+        images = sorted(images,
+                        key=lambda x: int(x.split('/')[-1].split('.')[0]))
+        for img in images:
+            frame = cv.imread(img)
+            yield frame
+
+def main():
+    """ Sample SiamRPN Tracker
+    """
+    parser = argparse.ArgumentParser(description='Use this script to run SiamRPN++ Visual Tracker',
+                                     formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument('--input_video', type=str, required=True, help='Path to input video for tracking')
+    parser.add_argument('--backbone_target', type=str, default='resnet_target.onnx', help='Path to ResNet ONNX model for target')
+    parser.add_argument('--backbone_search', type=str, default='resnet_search.onnx', help='Path to ResNet ONNX model for search')
+    parser.add_argument('--neck_1', type=str, default='neck_1.onnx', help='Path to Adjusted Layer 1 ONNX model')
+    parser.add_argument('--neck_2', type=str, default='neck_2.onnx', help='Path to Adjusted Layer 2 ONNX model')
+    parser.add_argument('--rpn_head', type=str, default='rpn_head.onnx', help='Path to RPN Head ONNX model')
+    args, _ = parser.parse_known_args()
+
+    backbone_search = cv.dnn.readNetFromONNX(args.backbone_search)
+    backbone_target = cv.dnn.readNetFromONNX(args.backbone_target)
+    neck_1 = cv.dnn.readNetFromONNX(args.neck_1)
+    neck_2 = cv.dnn.readNetFromONNX(args.neck_2)
+    rpn_head = cv.dnn.readNetFromONNX(args.rpn_head)
```

Comment:
Please check that model files exist

Reply:
I have updated my file to contain os. path.isfile to check the existence of the file in the relevant paths